hathor/transaction/scripts/opcode.py

```diff
@@ -591,21 +591,25 @@ def op_checkmultisig(context: ScriptContext) -> None:
     # For each signature we check if it's valid with one of the public keys
     # Signatures must be in order (same as the public keys in the multi sig wallet)
     pubkey_index = 0
+    old_stack = context.stack
     for signature in signatures:
         while pubkey_index < len(pubkeys):
             pubkey = pubkeys[pubkey_index]
             new_stack = [signature, pubkey]
-            op_checksig(ScriptContext(stack=new_stack, logs=context.logs, extras=context.extras, settings=settings))
+            context.stack = new_stack
+            op_checksig(context)
```

Comment:
I guess that's the bug fix, right? I'd like more information about the bug, its origin, and the fix.